kernel/src/process.rs

```diff
@@ -901,8 +901,11 @@ impl<C: Chip> ProcessType for Process<'_, C> {
                 },
                 _ => true,
             });
+            let count_after = tasks.len();
+            for _ in 0..count_before - count_after {
+                self.kernel.decrement_work();
```

Comment:
Is it feasible to just call this within the closure passed to `retain()`, whenever that closure returns true? It seems like a shame to include another iteration through the callbacks ring buffer when its not really needed. In fact, it seems like calculating `count_before` probably should have previously been gated by `CONFIG.trace_syscalls`

Reply:
Modified it, that is a good idea.